qregister.cpp

```diff
@@ -2706,6 +2707,60 @@ void CoherentUnit::QFT(bitLenInt start, bitLenInt length)
     }
 }
 
+/// "Entangled Hadamard" - perform an operation on two entangled registers like a bitwise Hadamard on a single
+/// unentangled register.
+void CoherentUnit::EntangledH(bitLenInt targetStart, bitLenInt entangledStart, bitLenInt length)
+{
+    std::unique_ptr<Complex16[]> nStateVec(new Complex16[maxQPower]);
+    std::fill(&(nStateVec[0]), &(nStateVec[0]) + maxQPower, Complex16(0.0, 0.0));
+    bitCapInt i, j, k;
+    bitLenInt len;
+    bool isOdd;
+    bitCapInt lengthPower = (1 << length);
+    bitCapInt lengthMask = lengthPower - 1;
+    bitCapInt targetMask = lengthMask << targetStart;
+    bitCapInt otherMask = (maxQPower - 1) & (~targetMask);
+    bitCapInt rowInt, colInt, interInt, otherRes;
+
+    const Complex16 zeroComplex(0.0, 0.0);
+
+    // As if multiplying a matrix times the state vector, we want to interate over every element of the array to
+    // multiply the state vector.
+    for (i = 0; i < maxQPower; i++) {
+        // If i points to a state with zero probability, the body is unnecessary.
+        if (stateVec[i] != zeroComplex) {
```

Comment:
General style suggestions - when you have full occlusion conditionals like this, it's often cleaner to use a `continue` instead.  That reduces the indentation one notch and removes one thing that the code reader needs to remember when reading the code.

No need to change unless you feel like it, but just a suggestion for the future.